module/zfs/vdev_removal.c

```diff
@@ -1780,6 +1785,8 @@ spa_vdev_remove_cancel_impl(spa_t *spa)
 		spa_config_enter(spa, SCL_ALLOC | SCL_VDEV, FTAG, RW_WRITER);
 		vdev_t *vd = vdev_lookup_top(spa, vdid);
 		metaslab_group_activate(vd->vdev_mg);
+		ASSERT(!vd->vdev_islog);
+		metaslab_group_activate(vd->vdev_log_mg);
```

Comment:
Again I may be wrong but wouldn't all cases of `metaslab_group_{activate,passivate}(vdev_log_mg)` that don't check if `vdev_log_mg != NULL` blow up for devices with less than 64 metaslabs?

Reply:
As you noted above, all non-log vdevs have a vdev_log_mg, even if it has no metaslabs in it.  Therefore this code path doesn't need to check if there's a vdev_log_mg to activate.